### PR TITLE
DAOS-13120 chk: properly handle check corpc

### PR DESCRIPTION
**Pull request by Nasf-Fan**
The middle node in the CaRT topo tree needs to properly aggregate the corpc reply from child nodes which can be leaf nodes, but also can be other middle nodes, and then forward related aggregated result to its parent node. Different from root node, the middle node cannot access the data structure resided on the check leader or PS leader, so need different aggregation logic.

Register co_post_reply to automatically handle those delay released buffer in related corpc reply that is allocated inside related corpc handler, that makes the logic more clean.

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by Nasf-Fan** [comment]
The rebuild timeout is not related with the CR logic patch.

**Comment by Nasf-Fan** [comment]
Ping reviewers, thanks!

**Review by ryon-jensen — APPROVED** [review]
LGTM

**Review by liuxuezhao — APPROVED** [review]

File: src/chk/chk_internal.h [hunk 1]
```diff
@@ -104,7 +105,9 @@ CRT_RPC_DECLARE(chk_start, DAOS_ISEQ_CHK_START, DAOS_OSEQ_CHK_START);
 #define DAOS_OSEQ_CHK_STOP							\
 	((int32_t)		(cso_status)		CRT_VAR)		\
 	((int32_t)		(cso_child_status)	CRT_VAR)		\
```
[the remark this answers is not in the document]
Comment (Nasf-Fan) [reply src/chk/chk_internal.h:107]: Originally, the cso_status is used to indicate whether the root itself executed local RPC succeed or not? if yes, then need to explicitly call post_reply to free some buffer. cso_child_status is used for others executing results. But it seems that we can call post_reply unconditionally, then do not need to distinguish local or remote failure.

**Review by liuxuezhao — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**